runtime/JavaScript/src/antlr4/TokenSource.d.ts

```diff
@@ -1,3 +1,7 @@
-export declare class TokenSource {
```

Comment:
mmm... not sure how this can work... the JS TokenSource does not implement these fields and methods.
I'd suggest making it a TS interface implemented by Lexer.
please bring this to a dedicated PR

Reply:
Ok, will do. For this PR I'll put BufferedTokenStream back to "tokenSource: Lexer", that was type safe though limiting.